sdk/appconfiguration/Azure.Data.AppConfiguration/src/SettingSelector.cs

```diff
@@ -48,27 +46,190 @@ public class SettingSelector
+        /// </summary>
+        public SettingSelector() { }
+
+        /// <summary>
+        /// Creates a <see cref="SettingSelector"/> that will retrieve <see cref="ConfigurationSetting"/> entities that match the passed-in key filter.
         /// </summary>
-        public SettingSelector() : this(Any, Any) { }
+        /// <param name="keyFilter">A key filter indicating which <see cref="ConfigurationSetting"/> entities to select.</param>
+        public static SettingSelector FromFilterString(string keyFilter)
+        {
+            Argument.AssertNotNullOrEmpty(keyFilter, nameof(keyFilter));
+            return new SettingSelector
+            {
+                _keys = new StringBuilder(keyFilter)
+            };
+        }
 
         /// <summary>
-        /// Creates a <see cref="SettingSelector"/> that will retrieve <see cref="ConfigurationSetting"/> entities that match
-        /// the passed-in keys and labels.
+        /// Creates a <see cref="SettingSelector"/> that will retrieve <see cref="ConfigurationSetting"/> entities that match the passed-in key filter.
         /// </summary>
-        /// <param name="key">A key or key filter indicating which <see cref="ConfigurationSetting"/> entities to select.</param>
-        /// <param name="label">A label or label filter indicating which <see cref="ConfigurationSetting"/> entities to select</param>
-        public SettingSelector(string key, string label = default)
+        /// <param name="keyFilter">A key filter indicating which <see cref="ConfigurationSetting"/> entities to select.</param>
+        /// <param name="labelFilter">A label filter indicating which <see cref="ConfigurationSetting"/> entities to select.</param>
+        public static SettingSelector FromFilterString(string keyFilter, string labelFilter)
         {
-            Keys = new List<string>
+            Argument.AssertNotNullOrEmpty(keyFilter, nameof(keyFilter));
+            Argument.AssertNotNullOrEmpty(labelFilter, nameof(labelFilter));
+            return new SettingSelector
             {
-                key ?? Any
+                _keys = new StringBuilder(keyFilter),
+                _labels = new StringBuilder(labelFilter)
             };
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public SettingSelector AddKey(string key)
+        {
+            Argument.AssertNotNullOrEmpty(key, nameof(key));
+            AppendSeparator(Keys);
+            AppendEscaped(Keys, key);
+            return this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public SettingSelector AddKeyStartsWith(string key)
```

Comment:
These Add* methods are really nice and I like them, but with the FromFilterString method, we enable this functionality.  Given that we don't know whether customers outside the provider will need these methods, and they are additive, I'd like to keep things simple for this iteration of the API.  Is there a strong reason not to use the Key and Labels properties we had previously?

Reply:
`Keys` must be usable after `SettingSelector` is created with `FromFilterString` method. So filter string must be parsed into a list of keys, but escaped `*` can't be represented in the `Keys`. Same with `Labels`.